UDPClientGUI/ViewModels/LoginViewModel.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Caliburn.Micro;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UDPCommGUI.ViewModels
+{
+    class LoginViewModel : Screen
+    {
+        private IEventAggregator _eventAggregator;
+        private string _username;
+        private string _password;
+
+        public LoginViewModel(IEventAggregator eventAggregator)
+        {
+            _eventAggregator = eventAggregator;
+            _eventAggregator.Subscribe(this);
+        }
+
+        public string Username
+        {
+            get => _username;
+            set
+            {
+                _username = value;
+                NotifyOfPropertyChange(nameof(Username));
+            }
+        }
+
+        public string Password
+        {
+            get => _password;
+            set
+            {
+                _password = value;
+                NotifyOfPropertyChange(nameof(Password));
+            }
+        }
+
+        public bool CanLogin(string username, string password)
+        {
+            if (username is null || username == "") return false;
```

Comment:
Use string.IsNullOrEmpty; and get into the habit of using trim() as a lot of strings have spaces at the start/end. All username/password fields shouldn't allow spaces.